packages/compiler-dom/src/transforms/vShow.ts

```diff
@@ -1 +1,47 @@
-// TODO
+import {
+  createStructuralDirectiveTransform,
+  createCompilerError,
+  ErrorCodes,
+  createSimpleExpression,
+  SimpleExpressionNode,
+  NodeTypes,
+  createConditionalExpression
+} from '@vue/compiler-core'
+
+export const transformShow = createStructuralDirectiveTransform(
+  'show',
+  (node, dir, context) => {
+    if (!dir.exp || !(dir.exp as SimpleExpressionNode).content.trim()) {
+      const loc = dir.exp ? dir.exp.loc : node.loc
+      context.onError(
+        createCompilerError(ErrorCodes.X_V_SHOW_NO_EXPRESSION, dir.loc)
+      )
+      dir.exp = createSimpleExpression(`true`, false, loc)
```

Comment:
Without an expression value, `v-show` should behave like an HTML boolean attribute i.e. equivalent to `v-show="true"`. So we can emit an error and abort further processing.